bundle/src/main/java/com/adobe/acs/commons/wcm/impl/AemEnvironmentIndicatorFilter.java

```diff
@@ -328,10 +352,14 @@ protected final void activate(ComponentContext ctx) {
                     "(" + HttpWhiteboardConstants.HTTP_WHITEBOARD_CONTEXT_NAME + "=*)");
             filterRegistration = ctx.getBundleContext().registerService(Filter.class.getName(),
                     this, filterProps);
+            
+            // Register the innerFilter so it is invoked after the WcmRequestFilter (Ranking = 2000)
```

Comment:
Can't we use declarative services for the inner filter as well and just depend on the outer one (by adding a reference to it)?

Reply:
It's also possible to remove the whole inner filter and just move reading the WCMMode after `filterChain.doFilter()`

Reply:
that sounds much better than the current approach, I'll give it a try.